src/serde_helpers.rs

```diff
@@ -0,0 +1,226 @@
+//! Collection of helper functions for serializing to and deserializing from BSON using Serde
+
+use std::{convert::TryFrom, result::Result};
+
+use serde::{de, ser, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{oid::ObjectId, Bson};
+
+/// Attempts to serialize a u32 as an i32. Errors if an exact conversion is not possible.
+pub fn serialize_u32_as_i32<S: Serializer>(val: &u32, serializer: S) -> Result<S::Ok, S::Error> {
+    match i32::try_from(*val) {
+        Ok(val) => serializer.serialize_i32(val),
+        Err(_) => Err(ser::Error::custom(format!("cannot convert {} to i32", val))),
+    }
+}
+
+/// Serializes a u32 as an i64.
+pub fn serialize_u32_as_i64<S: Serializer>(val: &u32, serializer: S) -> Result<S::Ok, S::Error> {
+    serializer.serialize_i64(*val as i64)
+}
+
+/// Attempts to serialize a u64 as an i32. Errors if an exact conversion is not possible.
+pub fn serialize_u64_as_i32<S: Serializer>(val: &u64, serializer: S) -> Result<S::Ok, S::Error> {
+    match i32::try_from(*val) {
+        Ok(val) => serializer.serialize_i32(val),
+        Err(_) => Err(ser::Error::custom(format!("cannot convert {} to i32", val))),
+    }
+}
+
+/// Attempts to serialize a u64 as an i64. Errors if an exact conversion is not possible.
+pub fn serialize_u64_as_i64<S: Serializer>(val: &u64, serializer: S) -> Result<S::Ok, S::Error> {
+    match i64::try_from(*val) {
+        Ok(val) => serializer.serialize_i64(val),
+        Err(_) => Err(ser::Error::custom(format!("cannot convert {} to i64", val))),
+    }
+}
+
+pub mod chrono_datetime_to_bson_datetime {
```

Comment:
I think instead of `thing1_to_thing2`, we should name these something that works for both serialization and deserialization. Maybe `thing1_as_thing2`?